source/kernel/bank2/fat.mac

```diff
@@ -869,6 +880,25 @@ zero_dsec_loop:	ld	(hl),c			; the data area of buffer
 ; Corrupts:    AF,BC,DE
 ;
 ;
+		;Update UD_ACLU to the number of the first cluster to be freed -1,
+		;but only if the new value would be smaller than the current value.
+		;This is to ensure that UD_ACLU is always either 1 or the
+		;lower free cluster number available -1.
+
+		push de
+		pop bc			;BC = first cluster to free (new value+1)
+		push hl
+		call ACLU_GET	;DE = current value of UD_ACLU
+		push bc
+		pop hl          ;HL = first cluster to free (new value+1)
+		ex de,hl        ;HL = current value, DE = new value+1
+		dec de          ;DE = new value
+		or a
+		sbc hl,de       ;HL = current value - new value, Cy=1 if new > current
+		pop hl          ;HL = address of unit descriptor

```

Comment:
I think these lines can be optimized as:
```
		push de		;SP = First cluster number in chain
		call ACLU_GET	;DE = current value of UD_ACLU
		ex (sp), hl     ;HL = first cluster to free (new value+1), SP = Address of unit descriptor
		ex de,hl        ;HL = current value, DE = new value+1
		dec de          ;DE = new value
		or a
		sbc hl,de       ;HL = current value - new value, Cy=1 if new > current
		pop hl          ;HL = address of unit descriptor
```
Gaining 26 t-states and 3 bytes.

Reply:
For some reason my brain has a difficult time handling the `ex (sp),hl` tricks so I don't use these often, but that optimization looks great, thanks!